### PR TITLE
piv: add read-object and write-object commands

### PR DESCRIPTION
**Pull request by dagheyman**
Add commands for writing and reading arbitrary PIV objects. The read commands should prompt for the pin if needed the write command should prompt for pin or management key if needed.

requested in #183

### CODE REVIEW

File: test/on_yubikey/cli_piv/test_misc.py [hunk 1]
```diff
@@ -11,3 +10,11 @@ def test_info(self):
...
         output = ykman_cli('piv', 'reset', '-f')
         self.assertIn('Success!', output)
+
+    def test_write_read_object(self):
+        ykman_cli(
+            'piv', 'write-object', '--id',
+            '0x5f0001', '-m', DEFAULT_MANAGEMENT_KEY,
+            '-', input='test data')
+        output = ykman_cli('piv', 'read-object', '--id', '0x5f0001')
+        self.assertIn('test data', output)
```
Comment (emlun) [inline test/on_yubikey/cli_piv/test_misc.py]: Why not `assertEqual[s]`?

File: ykman/cli/piv.py [hunk 2]
```diff
@@ -792,6 +792,68 @@ def unblock_pin(ctx, puk, new_pin):
             show_default=False, hide_input=True, err=True)
     controller.unblock_pin(puk, new_pin)
 
+@piv.command('read-object')
+@click_pin_option
+@click.pass_context
+@click.option(
+        '-i', '--id', 'object_id',
+        callback=lambda ctx, param, value: int(value, 16),
+        metavar='HEX', help='Id of PIV object.', required=True)
```
Comment (emlun) [inline ykman/cli/piv.py]: Why not make `--id` a required positional parameter instead?

File: ykman/cli/piv.py [hunk 3]
```diff
@@ -792,6 +792,68 @@ def unblock_pin(ctx, puk, new_pin):
...
+
+@piv.command('write-object')
+@click_pin_option
+@click_management_key_option
+@click.pass_context
+@click.option(
+        '-i', '--id', 'object_id',
+        callback=lambda ctx, param, value: int(value, 16),
+        metavar='HEX', help='Id of PIV object.', required=True)
+@click.argument('data', type=click.File('rb'), metavar='DATA')
```
Comment (emlun) [inline ykman/cli/piv.py]: It isn't clear from the help text that this should be a file name, or that `-` resolves to standard input.

File: ykman/cli/piv.py [hunk 4]
```diff
@@ -792,6 +792,68 @@ def unblock_pin(ctx, puk, new_pin):
...
+    do_read_object()
+
+@piv.command('write-object')
+@click_pin_option
+@click_management_key_option
+@click.pass_context
+@click.option(
+        '-i', '--id', 'object_id',
+        callback=lambda ctx, param, value: int(value, 16),
+        metavar='HEX', help='Id of PIV object.', required=True)
```
Comment (emlun) [inline ykman/cli/piv.py]: Why not make `--id` a required positional parameter instead?

**Comment by dagheyman** [comment]
Updated PR to address all comments. Ready for re-review!

**Pull request merged**